teps/0084-endtoend-provenance-collection.md

```diff
@@ -212,6 +489,32 @@ Go in to as much detail as necessary here.
 This might be a good place to talk about core concepts and how they relate.
 -->
 
+<!-- TODO: Revisit this once event-payload is taken into consideration -->
+
+**Multiple Attestations**: With this proposal, an image may have multiple attestations, one for the
+taskrun and one for the pipelinerun. When using the OCI storage, for example, the attestation image
+will contain 2 layers each representing one of the attestations. This is not necessarily a new
+behavior as the existing mechanism to attest taskruns could also produce multiple attestations for
+a single image under certain circumstances. However, this is much more likely to happen with this
+proposal. An example circumstance is if both the Task and the Pipeline provide the `IMAGE_URL` and
+`IMAGE_DIGEST` results.
```

Comment:
> This is not necessarily a new
behavior as the existing mechanism to attest taskruns could also produce multiple attestations for
a single image under certain circumstances.

By producing multiple attestations for a single image, do you mean multiple signatures (in the simplesigning attestation format) for a single image?

> An example circumstance is if both the Task and the Pipeline provide the `IMAGE_URL` and
`IMAGE_DIGEST` results.

IIUC, currently when Chains extracts image artifacts, it only [looks into the type hinting results at taskrun level](https://github.com/tektoncd/chains/blob/d4b02fef4a10634db20640930e7a46f68eedb389/pkg/artifacts/signable.go#L89-L127). If pipeline level also has the `IMAGE_URL` and `IMAGE_DIGEST` results, there will be no another simplesigning attestation generated for the image from pipeline level. Sorry if I misunderstand this.

Reply:
> By producing multiple attestations for a single image, do you mean multiple signatures (in the simplesigning attestation format) for a single image?

You quoted the bit about attestation, not signature. Not exactly sure what you mean. Currently, there are three sets of images involved. My terminology could be off, so let me state how I understand things for the sake of clarity:

1. Built Images: the images produced by the task.
2. Signature Images: images that hold a signature for a given built image. These are tagged as `sha256-xxx.sig`.
3. Attestation Images: images that hold one or more attestations for a given built image. Each attestation is a layer in the image. These are tagged as `sha256-xxx.att`.

What the "Multiple Attestations" section is trying to note is that there will likely be multiple layers for Attestation Images once this TEP is implemented. This is because Chains will first create the Attestation Image with a single layer containing the TaskRun attestation, then Chains will add a new layer to the Attestation Image containing the PipelineRun attestation.

I mention that this is not a new behavior because I could have two Tasks that produce type hinting results (e.g. `IMAGE_URL` and `IMAGE_DIGEST`). In that case, even without this TEP, the Attestation Image will contain two layers, each containing a different TaskRun attestation.

> IIUC, currently when Chains extracts image artifacts, it only [looks into the type hinting results at taskrun level](https://github.com/tektoncd/chains/blob/d4b02fef4a10634db20640930e7a46f68eedb389/pkg/artifacts/signable.go#L89-L127). If pipeline level also has the IMAGE_URL and IMAGE_DIGEST results, there will be no another simplesigning attestation generated for the image from pipeline level. Sorry if I misunderstand this.

This TEP makes Chains aware of type hinting on PipelineRun results.

Reply:
Thanks a lot for the clarification here. That's helpful!